File: java_solutions/2586.count-the-number-of-vowel-strings-in-range.java
/*
 * @lc app=leetcode id=2586 lang=java
 *
 * [2586] Count the Number of Vowel Strings in Range
 */

class Solution {
    public int vowelStrings(String[] words, int left, int right) {
       return Arrays.stream(words).collect(Collectors.toList()).subList(left,right+1).stream().filter(i-> "aeiou".contains(i.substring(0,1)) && "aeiou".contains(new StringBuilder(i).reverse().toString().substring(0,1))).collect(Collectors.toList()).size();
    }
}

File: kotlin_solutions/2586.count-the-number-of-vowel-strings-in-range.kt
/*
 * @lc app=leetcode id=2586 lang=kotlin
 *
 * [2586] Count the Number of Vowel Strings in Range
 */

class Solution {fun vowelStrings(words: Array<String>, left: Int, right: Int) = words.slice(left..right).count { it.first() in "aeiou" && it.last() in "aeiou" }}

